016-kafka-connect-build.md

```diff
@@ -0,0 +1,138 @@
+Then they have to build it and push into some container registry and configure the Kafka Connect deployment to use it.
+
+The `KafkaConnectS2I` is supported only on OKD / OpenShift and uses the S2I build to build the new container image with additional connector plugins.
+It is not support on Kubernetes since pure Kubernetes do not have any support for S2I builds.
+It makes it a bit easier to prepare the new image, but only slightly.
+User has to prepare a local directory with the desired connector plugins and pass it to the S2I build.
+The S2I build adds it to the image and uses it automatically in the Connect deployment.
+In order to add more connector plugins later, user needs to have the old connectors as well as the new connectors - it is not possible to just add the new connectors.
+
+Unlike other Strimzi components, none of these works in a declarative way.
+
+## Motivation
+
+One of the aims of Strimzi is to make using Kafka on Kubernetes as native as possible.
+A big part of that is being able to configure things in a declarative way.
+That is currently missing for adding connector plugins to the Connect deployments.
+
+Being able to configure the connector plugins in the `KafkaConnect` CR will make it easier to use for our users since they will not need to build the container images manually.
+It will also make it easier to have a connector catalog on the website since we will be able to just share the KafkaConnect CR including the build section to add the connector.
+
+## Proposal
+
+A new section named `build` would be added to the `KafkaConnect` custom resource.
+This section will allow users to configure:
+* connector plugins
+* build output
+
+The connector plugins section will be a list where one or more connectors can be defined.
+Each connector consists of a name and list of artifacts which should be downloaded.
+The artifacts can be of different types.
+For example:
+* `jar` for a directly download of Java JARs. 
+* `tgz` or `zip` to _download and unpack_ the artifacts. 
+The archives will be unpacked and checked for any possibly malicious content (e.g. symlinks).
+* `maven` to download JARs from Maven based on Maven coordinates. 
+User will provide `groupId`, `artifactId` and `version` and we will try to download the JAR as well as all its runtime dependencies.
+This might be also enhanced to support different Maven repositories etc.
+* and possibly others if required
+
+The second part - `output` - will configure how the newly built image will be handled.
+It will support two types:
+* `docker` to push the image to any Docker compatible container repository.
+* `imagestream` to push the image to OpenShift ImageStream (supported only on OpenShift)
+
+Strimzi will not run its own container repository.
+So users of the `docker` type build will need to provide credentials to their own Docker compatible container registry.
```

Comment:
I know it's in the following example but I think it's better to make it clear that it happens through a Secret that has to be referenced in the output section.